RedditSharp/Reddit.cs

```diff
@@ -317,6 +319,17 @@ public Comment GetComment(string subreddit, string name, string linkName)
             }
         }
 
+        public Listing<T> SearchByUrl<T>(string url) where T : Thing
+        {
+            var urlSearchQuery = string.Format(UrlSearchPattern, url);
+            return Search<T>(urlSearchQuery);
+        }
+
+        public Listing<T> Search<T>(string query) where T : Thing
```

Comment:
Not sure exactly what you need, but this should work just fine for a simple search:
var posts = new Reddit().Search<Post>("beautiful").GetLIsting().ToList();